app/controllers/two_factor_authentication/personal_key_verification_controller.rb

```diff
@@ -96,7 +96,6 @@ def handle_valid_otp
         redirect_to two_factor_options_url
       end
       reset_otp_session_data
-      user_session.delete(:mfa_device_remembered)
```

Comment:
We no longer need to delete these. Instead we will override `auth_method` with whatever the new method is.